src/odemis/acq/test/stream_test.py

```diff
@@ -2919,6 +3014,57 @@ def _roiToPhys(self, repst):
         logging.debug("Expecting pos %s, pxs %s, res %s", pos, pxs, res)
         return pos, pxs, res
 
+    def test_acq_arpol_live_update(self):
+        """
+        Test if live update works for AR
+        """
+        # Create the stream
+        sems = stream.SEMStream("test sem", self.sed, self.sed.data, self.ebeam)
+        # test when polarization analyzer hardware is present
+        ars = stream.ARSettingsStream("test ar with analyzer", self.ccd, self.ccd.data, self.ebeam, self.analyzer)
+
+        sas = stream.SEMARMDStream("test sem-ar", [sems, ars])
+
+        list_positions = list(ars.polarization.choices) + ["acquireAllPol"]
+
+        # to test each polarization position acquired sequentially in a single acq
+        ars.acquireAllPol.value = False
+
+        for pos in list_positions:
+            if pos == "acquireAllPol":
+                # to test all polarization position acquired in one acquisition
+                ars.acquireAllPol.value = True
+                # set pos to random pol pos from list as "acquireAllPol" is not a valid choice
+                pos = "vertical"
+
+            ars.polarization.value = pos
+
+            # Short acquisition (< 0.1s)
+            ars.integrationTime.value = 0.03  # s
+            # TODO use fixed repetition value -> set ROI?
+            ars.repetition.value = (1, 1)
+
+            # Start acquisition
+            # estimated acquisition time should be accurate with less than 50% margin + 1 extra second
+            timeout = 1 + 1.5 * sas.estimateAcquisitionTime()
+            f = sas.acquire()
+
+            # sas.raw: array containing as first entry the sem scan image for the scanning positions,
+            # rest are ar images
+            # data: array should contain same images as sas.raw
+
+            # Check if there is a live update in the setting stream.
+            time.sleep(2.0)
+            im1 = ars.image.value
+            time.sleep(3.0)
+            im2 = ars.image.value
+
+            # wait until it's over
+            data = f.result(timeout)
+
+            # Check if the image changed (live update is working)
+            self.assertRaises(AssertionError, numpy.testing.utils.assert_array_equal, im1, im2)
+
     def test_acq_arpol(self):
```

Comment:
Please also add a test for the `SEMTemporalMDStream`.

Reply:
Somehow the backend was not starting so I couldn't verify this test yet.